lib/shoppingcart.rb

```diff
@@ -0,0 +1,59 @@
+require './lib/shoppingcart'
+
+class ShoppingCart
+  attr_reader :name,
+              :capacity,
+              :products
+
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.to_i
+    @products = []
+    @details = nil
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    @details = {
+      name: @name,
+      capacity: @capacity
+    }
+  end
+
+  def is_full?
+    (@products.sum do |product|
+      product.quantity
+    end) > @capacity
+  end
+
+  def products_by_category(category)
+    @products.find_all do |product|
+      product.category == category
+    end
+  end
+
+  def percentage_occupied
+    ((@products.sum do |product|
+      product.quantity
+    end).to_f / @capacity) * 100
+  end
+
+  def sorted_products_by_quantity
+    @products.sort_by do |product|
+      product.quantity
+    end
+  end
+
+  def product_breakdown
+    product_breakdown= Hash.new
+    @products.each do |product|
+      product_breakdown[product.category] = products_by_category(product.category)
+    end
+    product_breakdown
+  end 
```

Comment:
Nice job here! I really like that you used the `products_by_category` method. If you wanted to take it a bit further, to make it a little cleaner, you could look into using the `group_by` enumerable.